Assets/VR4VET/Components/NPC/Scripts/AnimationS/NPCController.cs

```diff
@@ -33,17 +26,27 @@ void Update()
     {
         float distance = Vector3.Distance(target.position, transform.position);
         
-        // If the player is far away, go closer, but let them have some personal space. 
-        if (distance >= lookRadius) {
-            Vector3 direction = (target.position - transform.position).normalized;
-            agent.SetDestination(target.position - direction * personalSpaceFactor);
-            // we only walk forward
-            animator.SetFloat("VelocityY", agent.velocity.magnitude);
-        } else {
-            // when the navMeshAgent has angular speed 1200 (aka. turns instantly)
-            // we only walk forward
-            animator.SetFloat("VelocityY", agent.velocity.magnitude);
+        if (shouldFollow)
+        {
+            // If the player is far away, go closer, but let them have some personal space. 
+            if (distance >= lookRadius) 
+            {
+                Vector3 direction = (target.position - transform.position).normalized;
+                agent.SetDestination(target.position - direction * personalSpaceFactor);
+                animator.SetFloat("VelocityY", agent.velocity.magnitude);
+            }
+            else 
+            {
+                animator.SetFloat("VelocityY", 0); // Stop the NPC when close enough
```

Comment:
Change back to what it was before: 

Remove:
  animator.SetFloat("VelocityY", 0); // Stop the NPC when close enough
  agent.SetDestination(transform.position); // Stop the agent from moving
 
Add: 
  animator.SetFloat("VelocityY", agent.velocity.magnitude);